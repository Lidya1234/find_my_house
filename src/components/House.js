import React from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import '../style/style.css';
import '../style/Rating.css';

const House = ({
  id, name, image, price, rank,
}) => {
  const rate = (rank / 5) * 100;
  return (
    <Link to={{
      pathname: './SingleHouseList',
      state: { data: id },
    }}
    >
      <div className="house" data-testid="house">
        <img className="image" src={image} alt={image} />
        <div className="describe">
          <div className="name">
            {name}
            <div>
              {' '}
              <span className="star-wrapper">
                <span className="stars" style={{ width: `${rate}%` }} />
              </span>
            </div>
          </div>
          <div className="price">
            $
            {price}
            <div>per month</div>
          </div>

        </div>
      </div>
    </Link>
  );
};

House.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  rank: PropTypes.number.isRequired,
};
export default House;
