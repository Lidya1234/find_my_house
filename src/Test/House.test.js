import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../reducers/index';
import House from '../components/House';

afterEach(() => {
  cleanup();
});

describe('House component tests', () => {
  const house = {
    id: 1,
    name: 'Villa',
    image: 'https://res.cloudinary.com/lidu4394/image/upload/v1621343540/vu-anh-TiVPTYCG_3E-unsplash_1_gqha0y. jpg',
    price: '2000',
    rank: 4,
  };
  render(
    <Provider store={store}>
      <Router>
        <House house={House} />
      </Router>
    </Provider>,
  );
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <House house={house} />
        </Router>
      </Provider>,
    )
    .toJSON();

  const houseElement = screen.getByTestId('house');

  test('Component should Render', () => {
    expect(houseElement).toBeInTheDocument();
  });
  test('Component should contain a per month text', () => {
    expect(houseElement).toHaveTextContent('per month');
  });
  test('Matches snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
