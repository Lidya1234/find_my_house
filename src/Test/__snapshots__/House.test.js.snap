// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component tests Matches snapshot 1`] = `
<a
  href="./SingleHouseList"
  onClick={[Function]}
>
  <div
    className="house"
    data-testid="house"
  >
    <img
      className="image"
    />
    <div
      className="describe"
    >
      <div
        className="name"
      >
        <div>
           
          <span
            className="star-wrapper"
          >
            <span
              className="stars"
              style={
                Object {
                  "width": "NaN%",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="price"
      >
        $
        <div>
          per month
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`House component tests Matches snapshot 1`] = `
<a
  href="./SingleHouseList"
  onClick={[Function]}
>
  <div
    className="house"
    data-testid="house"
  >
    <img
      className="image"
    />
    <div
      className="describe"
    >
      <div
        className="name"
      >
        <div>
           
          <span
            className="star-wrapper"
          >
            <span
              className="stars"
              style={
                Object {
                  "width": "NaN%",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="price"
      >
        $
        <div>
          per month
        </div>
      </div>
    </div>
  </div>
</a>
`;
